src/state.py

```diff
@@ -200,32 +203,35 @@ class State(BaseModel):  # pylint: disable=too-few-public-methods
         proxy: proxy configuration.
     """
 
-    agent_password: typing.Optional[str] = None
+    agent_password: str | None = None
+    api_password: str
```

Comment:
Why don't we make this required and block the charm if it's not configured?

Reply:
We block. Empty values are not allowed and a validation error there will result in the charm reaching blocked status